### PR TITLE
Feat:(eos_designs): Support for NSSA in Router OSPF Configuration

### LINKED ISSUE #1061: Support for NSSA in Router OSPF Configuration
**Issue Type**

- [ ] Template enhancement
- [X ] Role enhancement
- [ ] Documentation enhancement (avd.sh)

**Is your feature request related to a problem? Please describe.**
Sort of, we are migrating to EVPN/VXLAN and some of our existing OSPF areas are NSSAs.
In order to migrate them easily, NSSA support would be great to have.

There is no structured data in the router ospf configuration on the cli config gen for NSSAs.

**Describe the solution you'd like**

It would be great to have the support for: 
- nssa nssa-only
-  nssa no summary
-  nssa default information originate 
- nssa lsa7 filter

**Describe alternatives you've considered**

add the following structured data as below (for example):

router_ospf:
      process_id:
        '1':
            nssa: (if nssa key exists, enable nssa e.g. area x nssa)
                nssa_only: <true | false>
                nssa_no_summary: '<true | false>
                default_information_originate: <true | false>

### PR DESCRIPTION
**Pull request by ankudinov**
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes)
- [ ] Documentation content changes
- [ ] Other (please describe):

## Related Issue(s)

Fixes #1061 

## Component(s) name

`arista.avd.eos_cli_config_gen`

## Proposed changes

Data model was extended to support OSPF stub and NSSA:
```yaml
router_ospf:
  process_ids:
    <process_id>:
      area_specific_config:
        - ospf_area: < ospf_area >
          stub: < no_summary | true | on >
        - ospf_area: < ospf_area >
          nssa: < no_summary | nssa_only | true | on >
          default_information_originate:
            metric: < Integer 1-65535 >  # Value of the route metric
            metric_type: < 1 | 2 >  # OSPF metric type
```

## How to test

Run molecule.

## Checklist:

- [x] My code has been rebased from devel before I start
- [x] I have read the [**CONTRIBUTING**](https://avd.sh/en/latest/docs/contributing/) document.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have updated [molecule CI](https://github.com/aristanetworks/ansible-avd/tree/devel/ansible_collections/arista/avd/molecule) testing accordingly
- [x] All new and existing tests passed ([`pre-commit`](https://www.avd.sh/docs/installation/development/#python-virtual-environment), `make linting` and `make sanity-lint`).

### CODE REVIEW

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/README.md [hunk 1]
```diff
@@ -2154,6 +2154,15 @@ router_ospf:
           summary_lsa:
             override_metric: < Integer 1-16777215 >
       mpls_ldp_sync_default: < true | false >
+      # area specific configuration
+      area_specific_config:
+        - ospf_area: < ospf_area >
+          stub: < no_summary | true | on >
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/README.md]: To be consistent with the rest of our data model, we should avoid mixed type variables. Also "on" is not used elsewhere.
I suggest:
```yaml
areas:
  < ospf_area >:
    type: < normal | stub | nssa | default -> normal >
    no_summary: < true | false >
    nssa_only: <true | false >
```

Comment (ankudinov) [reply ansible_collections/arista/avd/roles/eos_cli_config_gen/README.md]: done

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/documentation/router-ospf.j2 [hunk 2]
```diff
@@ -194,8 +194,8 @@
 
 ### Router OSPF Areas
 
-| Process ID | Area | Area Type | Filter Networks | Filter Prefix List | Additional Config Options |
-| ---------- | ---- | --------- | --------------- | ------------------ | ------------------------- |
+| Process ID | Area | Area Type | Filter Networks | Filter Prefix List | Additional Options |
+| ---------- | ---- | --------- | --------------- | ------------------ | ------------------ |
 {%         for process in router_ospf.process_ids | arista.avd.natural_sort %}
```
Comment (carlbuchmann) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/documentation/router-ospf.j2]: To follow the coding standard `for process` should be `for process_id`, I believe this was missed in the past, but let's correct this now to be consistent!

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/router-ospf.j2 [hunk 3]
```diff
@@ -27,6 +27,42 @@ router ospf {{ process_id }}
    bfd default
 {%     endif %}
 {%     for area in router_ospf.process_ids[process_id].areas | arista.avd.natural_sort %}
+{# OSPF stub area configuration #}
+{%         if router_ospf.process_ids[process_id].areas[area].type | arista.avd.default("normal") | lower == "stub" %}
```
Comment (carlbuchmann) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/router-ospf.j2]: In `eos_cli_config_gen` we typically don't normalize the string with  `| lower` - was there a specific reason in this use case?
I would have expected simply this: ` {%  if router_ospf.process_ids[process_id].areas[area].type | arista.avd.defined("stub") %}`

**Review by carlbuchmann — APPROVED** [review]
LGMT!

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/README.md [hunk 4]
```diff
@@ -2162,6 +2162,13 @@ router_ospf:
               - < IPv4 subnet / netmask >
               - < IPv4 subnet / netmask >
             prefix_list: < prefix list name >
+        < area >:
+          type: < normal | stub | nssa | default -> normal >
```
Comment (carlbuchmann) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/README.md:2166]: `normal` is not an option, this should should be `type: < stub | nssa >`

Comment (carlbuchmann) [reply ansible_collections/arista/avd/roles/eos_cli_config_gen/README.md:2166]: this should be just `type: < stub | nssa >

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/README.md [hunk 5]
```diff
@@ -2162,6 +2162,13 @@ router_ospf:
...
+          no_summary: < true | false >
+          nssa_only: <true | false >
```
Comment (carlbuchmann) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/README.md]: add space before `true` -> `nssa_only: < true | false >`

**Review by carlbuchmann — CHANGES_REQUESTED** [review]
Please see comments - updates required to documentation only.

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/README.md (hunk 5 above)
Comment (ankudinov) [reply ansible_collections/arista/avd/roles/eos_cli_config_gen/README.md]: done

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/README.md (hunk 4 above)
Comment (ankudinov) [reply ansible_collections/arista/avd/roles/eos_cli_config_gen/README.md:2166]: Using non-existing type default is not breaking. On the other hand it's good for readability to highlight that area type is not necessarily stub or nssa. I'd suggest to keep as is.

**Review by carlbuchmann — APPROVED** [review]
LGTM!

**Comment by ernestoherrerab** [comment]
Does this replace the below data structure? 

router_ospf:
  process_ids:
    < process_id >:
           areas:
             < area >:

**Pull request merged**